scio-core/src/main/scala/com/spotify/scio/values/SCollectionWithSideOutput.scala

```diff
@@ -32,19 +32,27 @@ import scala.jdk.CollectionConverters._
  * s of the [[SideOutput]] s are accessed via the additional [[SideOutputCollections]] return value.
  */
 class SCollectionWithSideOutput[T] private[values] (
-  val internal: PCollection[T],
-  val context: ScioContext,
```

Comment:
This is a breaking change.
We need to have access to the `Scollecction` to get the name. The new definition is consistent with other `SCollection` wrapper